source/codegen/validate_source_protos.py

```diff
@@ -0,0 +1,54 @@
+"""Script for validating the source proto files against generated ones.
+
+Expects input source path to source/codegen/protos and generated folder path.
+"""
+
+import argparse
+import filecmp
+import glob
+import os
+import sys
+
+
+def _check_mismatched_protos(source: str, generated: str) -> bool:
+    any_mismatches = False
+    file_paths = glob.glob(f"{source}/**/*.proto", recursive=True)
+    for file_path in file_paths:
+        driver_dir_name = os.path.split(os.path.split(file_path)[0])[1]
+        file_name = os.path.basename(file_path)
+        driver_and_file_path = f"{driver_dir_name}/{file_name}"
+        generated_file_path = f"{generated}/{driver_and_file_path}"
+        compare_result = filecmp.cmp(file_path, generated_file_path)
+        if not compare_result:
```

Comment:
When you have both `if` and `else` conditions, it's better to use positive logic rather than negative logic. In other words, flip the logic and the 2 blocks.

Reply:
File name changed to validate_imported_protos.py but I inverted the `if` and `else` to have positive logic.